rpc/client.go

```diff
@@ -109,6 +148,106 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+// getSignedUrl calls the Data Pipes API to retrieve a signed URL, where it can PUT the report json.
+// Data Pipes docs: https://github.com/10gen/data-pipes
+func getSignedURL(task string, execution int, data []byte, AWSAccessKey string, AWSSecretKey string, AWSToken string, dataPipesHost string, dataPipesRegion string) (string, error) {
+	service := "execute-api"
+	resultType := "cedar-report"
+	if AWSAccessKey == "" || AWSSecretKey == "" || dataPipesHost == "" || dataPipesRegion == "" {
+		return "", errors.New("Getting signed URL failed. AWS access key, AWS secret key, data pipes host and data pipes region required.")
```

Comment:
Can you use an error catcher here and maybe create a separate validate function? You could even call that validate func in the very beginning of `UploadReport` so you don't waste time if this is not configured correctly:
```
func (opts *UploadReportOptions) validate() error {
        catcher := grip.NewBasicCatcher()

        catcher.NewWhen(AWSAccessKey == "", "must provide an AWS access key")
        catcher.NewWhen(AWSSecretKey  == "", "must provide an AWS secret key")
        catcher.NewWhen(dataPipesHost  == "", "must provide the Data Pipes service hostname")
        catcher.NewWhen(dataPipesRegion  == "", "must provide the Data Pipes service region")

        return errors.Wrap(catcher.Resolve(), "invalid upload report options")
}

Reply:
I think you missed this one.

Reply:
Ah yeah forgot the delete